src/main/java/graphql/schema/impl/MultiReadOnlyGraphQLTypeVisitor.java

```diff
@@ -0,0 +1,205 @@
+package graphql.schema.impl;
+
+import graphql.Assert;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLCompositeType;
+import graphql.schema.GraphQLDirective;
+import graphql.schema.GraphQLDirectiveContainer;
+import graphql.schema.GraphQLEnumType;
+import graphql.schema.GraphQLEnumValueDefinition;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLFieldsContainer;
+import graphql.schema.GraphQLInputFieldsContainer;
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLInputType;
+import graphql.schema.GraphQLInterfaceType;
+import graphql.schema.GraphQLList;
+import graphql.schema.GraphQLModifiedType;
+import graphql.schema.GraphQLNonNull;
+import graphql.schema.GraphQLNullableType;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLOutputType;
+import graphql.schema.GraphQLScalarType;
+import graphql.schema.GraphQLSchemaElement;
+import graphql.schema.GraphQLTypeReference;
+import graphql.schema.GraphQLTypeVisitor;
+import graphql.schema.GraphQLUnionType;
+import graphql.schema.GraphQLUnmodifiedType;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+
+import java.util.List;
+
+public class MultiReadOnlyGraphQLTypeVisitor implements GraphQLTypeVisitor {
```

Comment:
Allows us to call N visitors at once - read only ones that can change nodes

Reply:
Is that public api?

Reply:
I am not so sure about the semantics here: they don't compose completely, right? I can't return a ABORT/Quit anymore, for example.

Reply:
They dont compose generically  - they run one after the other.  I have made it `@Internal` and improved the javadoc on what it does